### PR TITLE
feat: migrate from create-react-app to vite

### PR DESCRIPTION
**Pull request by SarjuHansaliya**
## Summary

* fix: update typescript
* fix: update sentry
* fix: update react-router
* fix: migrate to vite

### CODE REVIEW

File: .env.sample [hunk 1]
```diff
@@ -1,7 +1,4 @@
...
-REACT_APP_NETWORK_ID="1"
-REACT_APP_NETWORK_URL="https://api.avax.network/ext/bc/C/rpc"
-REACT_APP_MOONPAY_PK=""
-REACT_APP_COINBASE_PK=""
-REACT_APP_SUBGRAPH_BASE_URL="https://api.thegraph.com/subgraphs/name/pangolindex"
-REACT_APP_MIXPANEL=""
\ No newline at end of file
+VITE_MOONPAY_PK=""
+VITE_COINBASE_PK=""
+VITE_SUBGRAPH_BASE_URL="https://api.thegraph.com/subgraphs/name/pangolindex"
```
Comment (HariSeldon23) [inline .env.sample:3]: Could we move these URL's to the SDK repository?

File: src/pages/App.tsx [hunk 2]
```diff
@@ -54,117 +52,84 @@ interface IRoute {
...
 }
 
 export default function App() {
   const chainId = useChainId()
   const routes: IRoute[] = [
     {
       path: MENU_LINK.dashboard,
-      component: Dashboard,
-      layout: Layout
+      component: Dashboard
```
Comment (HariSeldon23) [inline src/pages/App.tsx:62]: How are we recording which features are live for which chains? Is this inferred from the data in the SDK?

File: src/pages/Dashboard/index.tsx [hunk 3]
```diff
@@ -23,9 +22,7 @@ const Dashboard = () => {
           <StatsWrapper isShowWatchList={WATCHLIST_ACCESS[chainId]}>
             <Portfolio />
 
-            {WATCHLIST_ACCESS[chainId] && (
-              <WatchList visibleTradeButton={true} tradeLinkUrl={MENU_LINK.swap} redirect={true} />
-            )}
+            {WATCHLIST_ACCESS[chainId] && <WatchList />}
```
Comment (HariSeldon23) [inline src/pages/Dashboard/index.tsx:25]: Now that the Watchlist isn't Chain dependent, we should actually show the watchlist in every chains Dashboard

File: .env.sample (hunk 1 above)
Comment (SarjuHansaliya) [reply .env.sample:3]: yes @HariSeldon23

File: src/pages/Dashboard/index.tsx (hunk 3 above)
Comment (SarjuHansaliya) [reply src/pages/Dashboard/index.tsx:25]: yes I will make that change

File: src/pages/App.tsx (hunk 2 above)
Comment (SarjuHansaliya) [reply src/pages/App.tsx:62]: as of now, its handled here `src/constants/accessPermissions.ts`

**Pull request merged**